zenith_utils/src/postgres_backend.rs

```diff
@@ -141,137 +152,158 @@ impl PostgresBackend {
         self.flush()
     }
 
-    // wrapper for run_internal() that shuts down socket when we are done
-    pub fn run(&mut self, handler: &mut impl Handler) -> Result<()> {
-        let ret = self.run_internal(handler);
+    // Wrapper for run_internal() that shuts down socket when we are done
+    pub fn run(mut self, handler: &mut impl Handler) -> Result<()> {
+        let ret = self.run_message_loop(handler);
         let _res = self.stream_out.shutdown(Shutdown::Both);
         ret
     }
```

Comment:
I changed `&mut self` to `mut self` to make it clear that the struct can't be reused after `run` has exited.